google-ads-examples/src/main/java/com/google/ads/googleads/examples/campaignmanagement/SetBidModifier.java

```diff
@@ -0,0 +1,135 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/** Sets a bid modifier for the mobile platform on given campaign. */
+public class SetBidModifier {
+
+  private static final float BID_MODIFIER = 1.5f;
+
+  private static class SetBidModifierParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.CAMPAIGN_ID, required = true)
+    private Long campaignId;
+  }
+
+  public static void main(String[] args) throws IOException {
+    SetBidModifierParams params = new SetBidModifierParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.campaignId = Long.parseLong("INSERT_CAMPAIGN_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new SetBidModifier().runExample(googleAdsClient, params.customerId, params.campaignId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param campaignId the campaign ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long campaignId) {
+    // Mobile device campaign criteria are unique in that all of the mobile device campaign
+    // criteria are created automatically at the time of campaign creation. Use the campaign ID and
+    // mobile device constant to get the resource name of a mobile device campaign criterion for a
+    // given campaign.
+    long mobileDeviceConstant = 30001;
```

Comment:
Instead of only making this work for mobile devices, how about making the criterion ID a parameter? That way, this can be reused by users for other use cases (locations, proximity, etc.).